client-lite/src/config/mcc_manager.h

```diff
@@ -11,14 +11,26 @@ class MCCManager
 public:
     MCCManager(ConfigManager& configManager);
 
-    std::string NextHost();
-    bool NoFallback() const;
-    bool ReportHostError(HRESULT hr, const std::string& host);
+    boost::optional<std::chrono::seconds> FallbackDelay();
+    std::string GetHost(const std::string& originalUrl);
+    void ReportHostError(HRESULT hr, UINT httpStatusCode, const std::string& mccHost, const std::string& originalUrl);
+    bool IsBanned(const std::string& mccHost, const std::string& originalUrl) const;
 
 private:
-    bool _IsFallbackDue(const std::string& host) const;
+    struct MccHost
```

Comment:
Generally, I see struct's being used as passive data structures ([https://en.wikipedia.org/wiki/Passive_data_structure](https://en.wikipedia.org/wiki/Passive_data_structure))

Since this contains features of OOP (access modifiers, member functions) - would it be better to have this as a class?

Reply:
Yes, will change to class.